src/pages/item-page/item-page.tsx

```diff
@@ -0,0 +1,83 @@
+import  { useEffect, useState } from 'react';
+import { Button, Icon, ItemContent, RecentlyViewed,  } from '@components';
+import { useLocation } from 'react-router-dom';
+import { ROUTES } from "@constants";
+import { useAppDispatch, useAppSelector } from "../../hooks/hooks";
+import { selectBook, fetchOneBook } from "@slices/oneBook-slice";
+import { addToFavorites, deleteFromFavorites, selectUserFavorites } from '@slices/user-slice';
+import { recentlyViewed } from "../../mock";
+
+
+  
+export const ItemPage: React.FC = () => {
+  const location = useLocation();
+  const books = location.state;
+  const dispatch = useAppDispatch();
+  const book = useAppSelector(selectBook);
+  const favorites = useAppSelector(selectUserFavorites);
+  const [isFavorite, setIsFavorite] = useState(
+    favorites.some((fav) => fav.key === books.key) || books?.isFavorite || false
+  );
+
+  const changeIcon = isFavorite ? 'bi-heart-fill' : 'bi-heart';
+  const addFavoriteButtonText = isFavorite ? 'Убрать из избранного' : 'Добавить в избранное'
+  const handleFavoriteClick = () => {
+    const bookData = {
+      key: books.key,
+      title: book.title,
+      author_name: books.author,
+      cover_edition_key: books.image,
+    };
+    if (isFavorite) {
+      dispatch(deleteFromFavorites(books.key));
+    } else {
+      dispatch(addToFavorites(bookData));
+    }
+    setIsFavorite(!isFavorite);
+  };
+
+  useEffect(() => {
+    dispatch(fetchOneBook(books.key));
+  }, [dispatch, books.key]);
+
+
+  const descriptionText =
+    typeof book.description === 'object' && book.description !== null
```

Comment:
чет замудрено
```suggestion
    book.description
```
так ведь тоже будет работать